src/shapes/image.class.js

```diff
@@ -373,10 +387,20 @@
      */
     setSrc: function(src, options) {
       var _this = this;
-      return fabric.util.loadImage(src, options).then(function(img) {
+      if (this.abortLoadingTask()) {
+        this.fire('loading:aborted');
+      }
+      var abortController = new AbortController();
+      this.__abortController = abortController;
```

Comment:
This is trickier
I think it should remain as is

Reply:
e.g. 
```js
img.set('src', someURL)
// very little time passed
img.set('src', someOtherURL)
```

Reply:
i don't think we should guard against this internally. I want to leave loading patterns to the developer. Give me some hour, my laptop is now dead, but i was trimming down this to show you i want to support abort signal.

Reply:
I understand what you mean.
I simply think we should support it as long as the loading task is part of the object's state.
Because it causes bugs that are hard to trace.
It reminds me the double init of canvas bug.
Concurrent bugs are unexpected.

Reply:
.